src/monitoring_tests/combinatorial_auction_surplus_test.py

```diff
@@ -0,0 +1,139 @@
+                        token_pair_objective_alt = Fraction(
+                            Fraction(solution_alt["objective"]["total"]), 10**18
+                        )
+
+                        a_abs_eth = (
+                            min(token_pair_surplus_alt, token_pair_objective_alt)
+                            - token_pair_surplus
+                        )
+                        a_rel = a_abs_eth / token_pair_surplus
+
+                        log_output = "\t".join(
+                            [
+                                "Combinatorial auction surplus test:",
+                                f"Tx Hash: {competition_data['transactionHash']}",
+                                f"Token pair: {token_pair}",
+                                f"Winning Solver: {solution['solver']}",
+                                f"Solver providing more surplus: {solution_alt['solver']}",
+                                f"Relative deviation: {float(a_rel * 100):.4f}%",
+                                f"Absolute difference: {float(a_abs_eth):.5f}ETH",
+                            ]
+                        )
+
+                        if (
+                            a_abs_eth > SURPLUS_ABSOLUTE_DEVIATION_ETH
+                            and a_rel > SURPLUS_REL_DEVIATION
+                        ):
+                            self.alert(log_output)
+                        elif (
+                            a_abs_eth > SURPLUS_ABSOLUTE_DEVIATION_ETH / 10
+                            and a_rel > SURPLUS_REL_DEVIATION / 10
+                        ):
+                            self.logger.info(log_output)
+                        else:
+                            self.logger.debug(log_output)
+
+        return True
+
+    def get_uid_trades(self, solution: dict[str, Any]) -> dict[str, Trade]:
+        """Get a dictionary mapping UIDs to trades in a solution."""
+        calldata = solution["callData"]
+        settlement = self.web3_api.get_settlement_from_calldata(calldata)
+        trades = self.web3_api.get_trades(settlement)
+        trades_dict = {
+            solution["orders"][i]["id"]: trade for (i, trade) in enumerate(trades)
+        }
+        return trades_dict
+
+    def get_token_pairs_surplus(
+        self, solution: dict[str, Any], prices: dict[str, float]
+    ) -> dict[tuple[str, str], Fraction]:
+        """Aggregate surplus of a solution on the different token pairs.
+        The result is a dict containing token pairs and the aggregated surplus on them.
+        """
+        trades_dict = self.get_uid_trades(solution)
+        surplus_dict: dict[tuple[str, str], Fraction] = {}
+        for uid in trades_dict:
+            trade = trades_dict[uid]
+            token_pair = (trade.data.sell_token.lower(), trade.data.buy_token.lower())
+
+            if trade.get_surplus_token() == trade.data.sell_token:
+                token_to_eth = Fraction(
```

Comment:
Can we make the name of the variable `token_to_eth`a bit more descriptive?

Reply:
Changed the name to `surplus_token_to_eth` and added a comment.